constantine/build.gradle

```diff
@@ -0,0 +1,166 @@
+plugins {
+    id 'java-library'
+    id 'maven-publish'
+    id 'com.jfrog.artifactory' version '4.20.0'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'net.java.dev.jna:jna:5.12.1'
+    testImplementation 'junit:junit:4.13.2'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+//task runNimbleMakeLib(type: Exec) {
+//    description = 'Runs CTT_LTO=false nimble make_lib inside constantine folder'
+//    workingDir 'constantine'
+//    environment 'CTT_LTO', 'false'
+//    commandLine 'nimble', 'make_lib'
+//}
+
+task compileJavaSource(type: Exec) {
+    description = 'Compiles the Java source files'
+    commandLine 'javac', 'src/main/java/org/hyperledger/besu/nativelib/constantine/LibConstantineEIP196.java'
+}
+
+task generateJNIHeader(type: Exec) {
+    description = 'Generates the JNI header file'
+    commandLine 'javac', '-h', 'constantine/constantine-jni/include', 'src/main/java/org/hyperledger/besu/nativelib/constantine/LibConstantineEIP196.java'
+}
+
+task compileNativeLibrary(type: Exec) {
+    description = 'Compiles the native library'
+    commandLine 'clang', '-I', "${System.env.JAVA_HOME}/include", '-I', "${System.env.JAVA_HOME}/include/darwin", '-shared', '-o', 'constantine-jni/lib/libconstantine.jnilib', 'constantine-jni/src/main/c/ethereum_evm_precompiles.c', '-I', 'constantine/include', '-I', 'src/main/java/org/hyperledger/besu/nativelib/constantine', '-L', 'constantine-jni/lib', '-lconstantine'
+}
```

Comment:
Idiomatic besu-native builds incorporate the build of the library into the root `build.sh` script and use automake or a similar tool to ensure that the appropriate toolchain is configured (cc, gcc, clang, etc).  

I think it is reasonable to use gradle for these tasks, but we should expect to be able to build on a variety of platforms.

Reply:
also, adding a `dependsOn` chain between these dependent tasks will make for an easier build ux.